src/rust/src/pkcs7.rs

```diff
@@ -164,6 +166,257 @@ fn encrypt_and_serialize<'p>(
     }
 }
 
+#[pyo3::pyfunction]
+fn decrypt_der<'p>(
+    py: pyo3::Python<'p>,
+    data: CffiBuf<'p>,
+    certificate: pyo3::Bound<'p, x509::certificate::Certificate>,
+    private_key: pyo3::Bound<'p, backend::rsa::RsaPrivateKey>,
```

Comment:
Sadly this can't take `backend::rsa::RsaPrivateKey`, we need to make sure we allow anything that implements the interface (including other Python classes), so this needs to be a `Bound<'p, PyAny>`

Reply:
OK! Does this mean the same for `pyo3::Bound<'p, x509::certificate::Certificate>` or not?

Reply:
Certificate is ok as-is.

Reply:
Perfect! I've added checks for RSA keys then. Thanks a lot!